File: python/20260317/allat.py
from __future__ import annotations

from datetime import datetime


class Allat:
    def __init__(self, nev: str, eletkor: int, bekerulesi_ev: int, orokbefogadhato: bool) -> None:
        self.nev = nev
        self.eletkor = eletkor
        self.bekerulesi_ev = bekerulesi_ev
        self.orokbefogadhato = orokbefogadhato

    def menhelyen_toltott_evek(self) -> int:
        return datetime.now().year - self.bekerulesi_ev

    def hangot_ad(self) -> str:
        raise NotImplementedError("A leszarmazott osztaly valositja meg.")

    def adatok_szovegesen(self) -> str:
        orokbefogadhato_szoveg = "igen" if self.orokbefogadhato else "nem"
        return (
            f"Nev: {self.nev}, eletkor: {self.eletkor} ev, bekerulesi ev: {self.bekerulesi_ev}, "
            f"orokbefogadhato: {orokbefogadhato_szoveg}, menhelyen toltott id: {self.menhelyen_toltott_evek()} ev"
        )

File: C#/20260317/Allatmenhely/Allat.cs
namespace Allatmenhely;

public abstract class Allat
{
    public string Nev { get; }
    public int Eletkor { get; }
    public int BekerulesiEv { get; }
    public bool Orokbefogadhato { get; }

    protected Allat(string nev, int eletkor, int bekerulesiEv, bool orokbefogadhato)
    {
        Nev = nev;
        Eletkor = eletkor;
        BekerulesiEv = bekerulesiEv;
        Orokbefogadhato = orokbefogadhato;
    }

    public int MenhelyenToltottEvek()
    {
        return DateTime.Now.Year - BekerulesiEv;
    }

    public abstract string HangotAd();

    public virtual string AdatokSzovegesen()
    {
        var orokbefogadhatoSzoveg = Orokbefogadhato ? "igen" : "nem";
        return $"Nev: {Nev}, eletkor: {Eletkor} ev, bekerulesi ev: {BekerulesiEv}, orokbefogadhato: {orokbefogadhatoSzoveg}, menhelyen toltott ido: {MenhelyenToltottEvek()} ev";
    }
}
